src/utils/validation.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * Created by rauliyohmc on 24/07/2017.
+ * @flow
+ */
+
+import type { Image, Options, Region, ColorProfile } from '../types';
+
+export const INVALID_IMAGE_MESSAGE = 'Invalid image param, you should either require a local asset, or provide an external URI';
+export const INVALID_OPTIONS_MESSAGE = 'Invalid options param, please refer to the API documentation';
+export const INVALID_POSITIVE_INTEGER_MESSAGE = 'Invalid maximumColorCount param, it should be a positive integer';
+export const INVALID_TYPE_MESSAGE = 'Invalid type, it should be a string';
+
+export default function validation(image: Image, options: Options) {
+  validateImage(image);
+  validateOptions(options);
+  if (options.type) {
+    validateType(options.type);
+  }
+  if (options.maximumColorCount) {
+    validateMaximumColorCount(options.maximumColorCount);
+  }
+  if (options.region) {
+    validateRegion(options.region);
+  }
+}
+
+export function validateImage(image: Image) {
+  if (typeof image !== 'number' && typeof image !== 'object') {
+    throw new Error(INVALID_IMAGE_MESSAGE);
+  }
+  if (typeof image === 'object') {
+    if (!image.uri) throw new Error(INVALID_IMAGE_MESSAGE);
+    if (typeof image.uri !== 'string') throw new Error(INVALID_IMAGE_MESSAGE);
+  }
+}
+
+export function validateOptions(options: Options) {
```

Comment:
This should be `validateOptionsKeys`, since `validateOptions` gives a hint that it will validate also a `type`, `maximumColorCount` and `region`, but it's done separately.